docsrc/conf.py

```diff
@@ -45,6 +45,7 @@
     'IPython.sphinxext.ipython_console_highlighting',
     'sphinx.ext.mathjax',
     "sphinx.ext.napoleon",
+    'sphinx_autodoc_typehints',
```

Comment:
I'm slightly confused what this extension actually does. As far as I can tell, if there is no type for a parameter in the docstring, `sphinx_autodoc` already adds the type?

Reply:
This module was added to address the link creation issue requested in previous review.

I don't know enough about documentation to give a definitive answer.